firedrake/assemble.py

```diff
@@ -169,239 +160,240 @@ def thunk():
     return thunk
 
 
-@utils.known_pyop2_safe
-def _assemble(f, tensor=None, bcs=None, form_compiler_parameters=None,
-              inverse=False, mat_type=None, sub_mat_type=None,
-              appctx={},
-              options_prefix=None,
-              assemble_now=False,
-              allocate_only=False,
-              zero_tensor=True,
-              diagonal=False):
-    r"""Assemble the form or Slate expression f and return a Firedrake object
-    representing the result. This will be a :class:`float` for 0-forms/rank-0
-    Slate tensors, a :class:`.Function` for 1-forms/rank-1 Slate tensors and
-    a :class:`.Matrix` for 2-forms/rank-2 Slate tensors.
+def get_matrix(form, mat_type, sub_mat_type, *, bcs=None,
+               options_prefix=None, tensor=None, appctx=None,
+               form_compiler_parameters=None):
+    matfree = mat_type == "matfree"
```

Comment:
Maybe add documentation?

Reply:
Some light documentation for all the new functions would be nice.

And this comment is a bit nitpicky but is it appropriate to call the first argument `form` when it's clearly not always a form? I realize we've done it this way for a long while, but it might be more accurate to simply replace `form` with `expr` here (and elsewhere when appropriate).

Reply:
Done both of those things.

Reply:
Done.